### PR TITLE
[FIX] web_editor: convert p into span in li

### PR DESCRIPTION
**Pull request by visp-odoo**
**Current behavior before PR:**

if you have p within li,sometimes removing the p will result in the loss of all classes.

**Desired behavior after PR is merged:**

Replace p inside li with span while preserving classes.

task-3546209

opw-3602047

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/src/js/editor/odoo-editor/test/spec/copyPaste.test.js [hunk 1]
```diff
@@ -333,6 +334,44 @@ describe('Paste', () => {
                     contentAfter: '<ul><li><h1>Google</h1></li><li><h1>Test</h1></li><li><h1>test2</h1></li></ul><p>[]<br></p>',
                 });
             });
+            it('should convert p into span if p has classes', async () => {
```
Comment (sebgeelen) [inline addons/web_editor/static/src/js/editor/odoo-editor/test/spec/copyPaste.test.js:337]: 1. Can you move this unit test from copy paste to sanitize please.
Your change is in `sanitize.js` so it do not make sense to test it with a unit test in `copyPaste.test.js`.

2. Your unit test is way too complicated, you should always write the smallest unit test possible that actuality test your change.
Something like : 

```
...
contentBefore: '<ul><li><p class="class-1">abc</p><p class="class-2">def</p></li></ul>',
contentAfter: '<ul><li><span class="class-1">abc</span><span class="class-2">def</span></li></ul>'
...
```

**Review by sebgeelen — CHANGES_REQUESTED** [review]

File: addons/web_editor/static/src/js/editor/odoo-editor/test/spec/copyPaste.test.js (hunk 1 above)
Comment (visp-odoo) [reply addons/web_editor/static/src/js/editor/odoo-editor/test/spec/copyPaste.test.js]: @sebgeelen Done changes. :+1:

**Comment by sebgeelen** [comment]
@odoo/rd-security Could I have an override please ?

Reason : replace an HtmlElement with another HtmlElement using `replaceWith()`, The new element is instantiated by us, and only contain node already in the document. No new content is added in this element.

**Comment by sebgeelen** [comment]
@robodoo r+

**Pull request closed without merging**